docs/installation.rst

```diff
@@ -1,11 +1,46 @@
 Installation
 ============
 
-To install this package run
+To use StarkNet.py, ``ecdsa, fastecdsa, sympy`` dependencies are required. Depending on the operating system,
+different install steps must be performed.
 
-``pip install starknet.py``
+Linux
+^^^^^
 
-or using Poetry:
+.. code-block:: console
 
-``poetry add starknet.py``
+    sudo apt install -y libgmp3-dev
+    pip install starknet-py
 
+MacOS with Intel processor
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+
+.. code-block:: console
+
+    brew install gmp
+    pip install starknet-py
+
+MacOS with Apple silicon
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Requires `Homebrew <https://brew.sh/>`_  installation.
+First install gmp:
+
+.. code-block:: console
+
+    brew install gmp
+
+Then install required dependencies and StarkNet.py itself:
+
+.. code-block:: console
+
+    CFLAGS=-I`brew --prefix gmp`/include LDFLAGS=-L`brew --prefix gmp`/lib pip install ecdsa fastecdsa sympy
+    pip install starknet-py
+
+Windows
+^^^^^^^
+
+This library is incompatible with Windows devices.
+Use virtual machine with Linux, Windows Subsystem for Linux 2 (WSL2) or other solution.
```

Comment:
WSL should be the only suggestion, it works well

Reply:
But using a VM is also a valid option. I specifically don't want to recommend a single solution so if someone has problems it doesn't look like "I used wsl2 as you said and it still doesn't work".

Reply:
yeah, I remember the dark times when I was forced to use windows, and cygwin was pretty ok too

Reply:
Using VM is a huge overkill for using a python library. Using docker makes more sense.

Reply:
this all falls into `or other solution`. If someone knows how to setup docker on windows (arguably harder than setting up a vm), they wouldn't need our recommendation anyways.